src/config/syscheck-config.c

```diff
@@ -2573,12 +2517,21 @@ static int process_option_regex(char *option, OSMatch ***syscheck_option, xml_no
 static void process_option(char ***syscheck_option, xml_node *node) {
 
     unsigned int counter_opt = 0;
+    char *real_path;
+    char *dir = strdup(node->content);
+    char *tmp_dir;
     char **new_opt = NULL;
 
+    tmp_dir = validate_path(dir);
+    if (tmp_dir == NULL) {
+        os_free(dir);
+        return;
+    }
+
     /* We attempt to expand environment variables */
-    if (new_opt = get_paths_from_env_variable(node->content), !new_opt) {
+    if (new_opt = get_paths_from_env_variable(tmp_dir), !new_opt) {
         os_calloc(2, sizeof(char *), new_opt);
-        os_strdup(node->content, new_opt[0]);
+        os_strdup(dir, new_opt[0]);
         new_opt[1] = NULL;
```

Comment:
I've seen this done a couple of times, where if `get_paths_from_env_variable` returns NULL, a suplementary buffer is reserved and used instead, could we move this inside `get_paths_from_env_variable` and have it guarantee that the retunr value is never NULL? It would simplify the code and improve readability.